Simplenote/src/androidTest/java/com/automattic/simplenote/utils/TestBucket.kt

```diff
@@ -39,7 +54,102 @@ abstract class TestBucket<T : Syncable>(name: String) : Bucket<T>(null, name, nu
         }
     }
 
+    override fun allObjects(): ObjectCursor<T> {
+        return TestObjectCursor(objects)
+    }
+
+    override fun reset() {
+        clear()
+    }
+
+    override fun query(): Query<T> {
+        return TestQuery(objects)
+    }
+
     fun clear() {
         objects.clear()
     }
+
+}
+
+class TestObjectCursor<T : BucketObject>(private val objects: MutableList<T>) : AbstractCursor(), Bucket.ObjectCursor<T> {
+    private val columns = arrayOf("simperiumKey", "object")
+
+    override fun getCount(): Int {
+        return objects.size
+    }
+
+
+    override fun getColumnNames(): Array<String> {
+        return columns
+    }
+
+    override fun getColumnCount(): Int {
+        return columns.size
+    }
+
+    override fun getBlob(columnIndex: Int): ByteArray {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getString(columnIndex: Int): String {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun copyStringToBuffer(columnIndex: Int, buffer: CharArrayBuffer?) {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getShort(columnIndex: Int): Short {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getInt(columnIndex: Int): Int {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getLong(columnIndex: Int): Long {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getFloat(columnIndex: Int): Float {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getDouble(columnIndex: Int): Double {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getType(columnIndex: Int): Int {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun isNull(column: Int): Boolean {
+        throw RuntimeException("not implemented")
+    }
+
+    override fun getSimperiumKey(): String {
+        return objects[position].simperiumKey
+    }
+
+    override fun getObject(): T {
+        return objects[position]
+    }
+}
+
+class TestQuery<T : BucketObject>(private val objects:  MutableList<T>) : Query<T>() {
```

Comment:
Praise (❤️): This is nicely node, kudos!

Reply:
Thanks!